### PR TITLE
krew: Add version 0.4.2

### LINKED ISSUE #1234: [Request] Add Krew plugin manager
I'd be nice to have Krew available. https://github.com/kubernetes-sigs/krew

### PR DESCRIPTION
**Pull request by jamezrin**
Closes #1234 

- [x] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).

### CODE REVIEW

File: bucket/krew.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    },
+    "installer": {
+        "script": [
+            "Invoke-ExternalCommand -Path \"$dir\\krew.exe\" -Args 'install krew' | Out-Null"
+        ]
+    },
+    "env_add_path": ".krew\\bin",
+    "persist": [
+        ".krew"
+    ],
```
Comment (rashil2000) [inline bucket/krew.json:24]:
```suggestion
    "persist": ".krew",
```

File: bucket/krew.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    },
+    "architecture": {
+        "64bit": {
+            "url": "https://github.com/kubernetes-sigs/krew/releases/download/v0.4.2/krew.exe",
+            "hash": "2729c298b5d90b531585939f080a3968c1a672e6223b4313eea8adb5396b4840"
+        }
+    },
+    "installer": {
+        "script": [
+            "Invoke-ExternalCommand -Path \"$dir\\krew.exe\" -Args 'install krew' | Out-Null"
```
Comment (rashil2000) [inline bucket/krew.json:18]:
```suggestion
            "[Environment]::SetEnvironmentVariable('KREW_ROOT', \"$persist_dir\\.krew\", 'Process')",
            "Invoke-ExternalCommand -Path \"$dir\\krew.exe\" -Args 'install krew' | Out-Null"
```

File: bucket/krew.json [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        ]
+    },
+    "env_add_path": ".krew\\bin",
```
Comment (rashil2000) [inline bucket/krew.json:21]:
```suggestion
    "env_add_path": ".krew\\bin",
    "env_set": {
        "KREW_ROOT": "$persist_dir\\.krew"
    },
```

**Review by rashil2000 — CHANGES_REQUESTED** [review]

File: bucket/krew.json [hunk 4]
```diff
@@ -0,0 +1,38 @@
+{
+    "version": "0.4.2",
+    "description": "Krew is the package manager for kubectl plugins.",
+    "homepage": "https://krew.sigs.k8s.io/",
+    "license": "Apache-2.0",
+    "notes": "This script needs to be executed as an administrator.",
+    "suggest": {
+        "gsudo": "gsudo"
+    },
```
Comment (rashil2000) [inline bucket/krew.json:9]:
```suggestion
    "notes": "This script needs to be executed as an administrator. You can install 'gsudo' to run it.",
```

It's only required during initial setup

**Review by rashil2000 — CHANGES_REQUESTED** [review]

**Comment by rashil2000** [comment]
Question: What all things does the `krew install` command do? I think we can bypass the admin requirement.

**Comment by jamezrin** [comment]
> Question: What all things does the `krew install krew` command do? I think we can bypass the admin requirement.

One of the things that it does is creating a symlink, for which you need admin privileges unless you have the developer mode of windows is enabled. Check this for more info https://github.com/kubernetes-sigs/krew/issues/378

File: bucket/krew.json [hunk 5]
```diff
@@ -0,0 +1,37 @@
+{
+    "version": "0.4.2",
+    "description": "Krew is the package manager for kubectl plugins.",
+    "homepage": "https://krew.sigs.k8s.io/",
+    "license": "Apache-2.0",
+    "notes": "This script needs to be executed as an administrator. You can install 'gsudo' to run it.",
```
Comment (rashil2000) [inline bucket/krew.json:6]:
```suggestion
    "notes": [
        "This application creates symlinks during installation of plugins.",
        "You can install 'gsudo' to run it as administrator, or enable 'Developer Mode' in Windows 10 Settings to bypass the elevation requirement."
    ],
```

**Review by rashil2000 — CHANGES_REQUESTED** [review]

File: bucket/krew.json (hunk 5 above)
Comment (rashil2000) [reply bucket/krew.json:6]: In that case it's better to clarify here.

**Pull request merged**